src/module/PropagationCK.cpp

```diff
@@ -79,36 +79,54 @@ void PropagationCK::process(Candidate *candidate) const {
 	ParticleState &current = candidate->current;
 	candidate->previous = current;
 
-	double step = clip(candidate->getNextStep(), minStep, maxStep);
+	Y yIn(current.getPosition(), current.getDirection());
+	double step = maxStep;
 
 	// rectilinear propagation for neutral particles
 	if (current.getCharge() == 0) {
-		Vector3d pos = current.getPosition();
-		Vector3d dir = current.getDirection();
-		current.setPosition(pos + dir * step);
+		step = std::max(minStep, candidate->getNextStep());
```

Comment:
same comment as before: unless I'm missing something, max() is not the same as clip().